facilitator/src/transport.rs

```diff
@@ -130,33 +476,339 @@ mod tests {
+                    // previous UploadPart failure
+                    MockRequestDispatcher::with_status(204)
+                        .with_request_checker(is_abort_multipart_upload_request),
+                    // Response to UploadPart. HTTP 200 but no ETag header will
+                    // cause failure.
+                    MockRequestDispatcher::with_status(200)
+                        .with_request_checker(is_upload_part_request),
+                    // Response to AbortMultipartUpload, expected because of
+                    // previous UploadPart failure
+                    MockRequestDispatcher::with_status(204)
+                        .with_request_checker(is_abort_multipart_upload_request),
+                    // Well formed response to UploadPart.
+                    MockRequestDispatcher::with_status(200)
+                        .with_request_checker(is_upload_part_request)
+                        .with_header("ETag", "fake-etag"),
+                    // Well formed response to UploadPart
+                    MockRequestDispatcher::with_status(200)
+                        .with_request_checker(is_upload_part_request)
+                        .with_header("ETag", "fake-etag"),
+                    // Well formed response to CompleteMultipartUpload
+                    MockRequestDispatcher::with_status(200)
+                        .with_request_checker(is_complete_multipart_upload_request)
+                        .with_body(
+                            r#"<?xml version="1.0" encoding="UTF-8"?>
+<CompleteMultipartUploadResult>
+   <Location>string</Location>
+   <Bucket>fake-bucket</Bucket>
+   <Key>fake-key</Key>
+   <ETag>fake-etag</ETag>
+</CompleteMultipartUploadResult>"#,
+                        ),
+                    // Well formed response to CompleteMultipartUpload
+                    MockRequestDispatcher::with_status(200)
+                        .with_request_checker(is_complete_multipart_upload_request)
+                        .with_body(
+                            r#"<?xml version="1.0" encoding="UTF-8"?>
+<CompleteMultipartUploadResult>
+   <Location>string</Location>
+   <Bucket>fake-bucket</Bucket>
+   <Key>fake-key</Key>
+   <ETag>fake-etag</ETag>
+</CompleteMultipartUploadResult>"#,
+                        ),
+                    // Failure response to CompleteMultipartUpload
+                    MockRequestDispatcher::with_status(400)
+                        .with_request_checker(is_complete_multipart_upload_request),
+                ];
+                S3Client::new_with(
+                    MultipleMockRequestDispatcher::new(requests),
+                    MockCredentialsProvider,
+                    region.clone(),
+                )
+            },
+        )
+        .expect("failed to create multipart upload writer");
+
+        // First write will fail due to HTTP 401
+        writer.write(&[0; 51]).expect_err("expected error");
```

Comment:
These messages are all quite superfluous. For tests just use `unwrap()` and `unwrap_err()`, the error will have the actual code line.